ledger/complete/wal/compactor_test.go

```diff
@@ -86,13 +110,15 @@ func Test_Compactor(t *testing.T) {
 				savedData[rootHash] = data
 			}
 
-			assert.Eventually(t, func() bool {
-				from, to, err := checkpointer.NotCheckpointedSegments()
-				require.NoError(t, err)
+			// wait for the bound-checking observer to confirm checkpoints have been made
+			for range co.done {
+				break
+			}
```

Comment:
Why this vs simply:

```go
<-co.done
```
?

Reply:
Also, I think it's great that we can avoid using `assert.Eventually` which basically just keeps polling the given condition, but perhaps we still want to keep the timeout here since the channel read could potentially block forever.

How about this:
```suggestion
			select {
			case <-co.done:
				// continue
			case <-time.After(20*time.Second):
			        assert.FailNow(t, "timed out")
			}
```

Reply:
For the record, the single-case select hits [S1000](https://github.com/dominikh/go-tools/blob/878d7b81f5e581159102230618782513c79c1fa0/simple/lint.go#L41) but the one with the timeout doesn't :)